### DISCUSSION TITLE
Add support for beet + bolt

### DISCUSSION THREAD

**Discussion by juice-ix** [IoeCmcomc/MCDatapacker#10] [Ideas]
Just as the title says. Just in case you don't know, beet is a basically a tool that helps with datapack workflow. And bolt is a beet extension that allows you to run python in datapacks. 
A regular datapack also works as a bolt datapack, so it doesn't require any special exceptions or anything. All you'd need to do is add support for python syntax when running beet and bolt together

**Comment by IoeCmcomc** [comment]
It seems like a good idea, but I haven't had any experience with these tools yet.
After investigating, I found some datapacks using `Beet` and `Bolt` ([here](https://github.com/ps-dps/MobCaptains) and [here](https://github.com/2mal3/2mal3s-Recipes)). They have a different folder structure from regular datapacks. Because MCDatapacker assumes all folders containing a `pack.mcmeta` file and a `data` sub-folder as datapacks, supporting _Beet_ projects is not trivial.
_Bolt_ seems like a custom language mixing _mcfunction_ and _Python_ syntaxes. Writing another parser for _Bolt_ will take time unless there is a way to get syntax errors directly from _Beet_. Adding syntax highlighting is easier, though.
At least _Beet_ can automatically rebuild files at changes, so it should already work with MCDatapacker, to some degree.
